xbmc/addons/AddonBuilder.h

```diff
@@ -30,36 +30,36 @@ class CAddonBuilder
+  void SetLicense(std::string license) { m_addonInfo.m_license = std::move(license); }
+  void SetSummary(std::string summary) { m_addonInfo.m_summary = std::move(summary); }
+  void SetDescription(std::string description) { m_addonInfo.m_description = std::move(description); }
+  void SetDisclaimer(std::string disclaimer) { m_addonInfo.m_disclaimer = std::move(disclaimer); }
+  void SetAuthor(std::string author) { m_addonInfo.m_author = std::move(author); }
+  void SetSource(std::string source) { m_addonInfo.m_source = std::move(source); }
+  void SetIcon(std::string icon) { m_addonInfo.m_icon = std::move(icon); }
+  void SetArt(std::string type, std::string value) { m_addonInfo.m_art[type] = value; }
+  void SetArt(std::map<std::string, std::string> art) { m_addonInfo.m_art = std::move(art); }
+  void SetScreenshots(std::vector<std::string> screenshots) { m_addonInfo.m_screenshots = std::move(screenshots); }
+  void SetChangelog(std::string changelog) { m_addonInfo.m_changelog = std::move(changelog); }
+  void SetBroken(std::string broken) { m_addonInfo.m_broken = std::move(broken); }
+  void SetPath(std::string path) { m_addonInfo.m_path = std::move(path); }
+  void SetLibName(std::string libname) { m_addonInfo.m_libname = std::move(libname); }
+  void SetVersion(AddonVersion version) { m_addonInfo.m_version = std::move(version); }
+  void SetMinVersion(AddonVersion minversion) { m_addonInfo.m_minversion = std::move(minversion); }
+  void SetDependencies(ADDONDEPS dependencies) { m_addonInfo.m_dependencies = std::move(dependencies); }
   void SetExtrainfo(InfoMap extrainfo) { m_addonInfo.extrainfo = std::move(extrainfo); }
-  void SetType(TYPE type) { m_addonInfo.type = type; }
+  void SetType(TYPE type) { m_addonInfo.m_mainType = type; }
   void SetExtPoint(cp_extension_t* ext) { m_extPoint = ext; }
-  void SetInstallDate(CDateTime installDate) { m_addonInfo.installDate = installDate; }
-  void SetLastUpdated(CDateTime lastUpdated) { m_addonInfo.lastUpdated = lastUpdated; }
-  void SetLastUsed(CDateTime lastUsed) { m_addonInfo.lastUsed = lastUsed; }
-  void SetOrigin(std::string origin) { m_addonInfo.origin = std::move(origin); }
-  void SetPackageSize(uint64_t size) { m_addonInfo.packageSize = size; }
+  void SetInstallDate(CDateTime installDate) { m_addonInfo.m_installDate = installDate; }
+  void SetLastUpdated(CDateTime lastUpdated) { m_addonInfo.m_lastUpdated = lastUpdated; }
+  void SetLastUsed(CDateTime lastUsed) { m_addonInfo.m_lastUsed = lastUsed; }
+  void SetOrigin(std::string origin) { m_addonInfo.m_origin = std::move(origin); }
+  void SetPackageSize(uint64_t size) { m_addonInfo.m_packageSize = size; }
 
-  const std::string& GetId() const { return m_addonInfo.id; }
-  const AddonVersion& GetVersion() const { return m_addonInfo.version; }
+  const std::string& GetId() const { return m_addonInfo.m_id; }
+  const AddonVersion& GetVersion() const { return m_addonInfo.m_version; }
```

Comment:
The add of `friend class CAddonBuilder` is only temporarily to allow them to set the values in CAddonInfo, on end create CAddonInfo the values itself by his constructor.